block/src/Pos/DB/Block.hs

```diff
@@ -94,11 +88,15 @@ getTipBlock = getTipSomething "block" getBlock
 getSerializedBlock
     :: forall ctx m. (HasConfiguration, MonadRealDB ctx m)
     => HeaderHash -> m (Maybe ByteString)
-getSerializedBlock = blockDataPath >=> getRawData
+getSerializedBlock hh = do
+    DB{..} <- getBlockDataDB
+    liftIO $ Rocks.get rocksDB rocksReadOpts (fromString . formatToString hashHexF $ hh)
 
 -- Get serialization of an undo data for block with given hash from Block DB.
 getSerializedUndo :: (HasConfiguration, MonadRealDB ctx m) => HeaderHash -> m (Maybe ByteString)
-getSerializedUndo = undoDataPath >=> getRawData
+getSerializedUndo hh = do
+    DB{..} <- getBlockDataDB
+    liftIO $ Rocks.get rocksDB rocksReadOpts (fromString . undoModifier . formatToString hashHexF $ hh)
```

Comment:
Normally we use keys which follow the following pattern: `a/b` where `a` denotes type of data or part of DB (e. g. update system part of GState DB) and `b` is serialized key corresponding to a value.
So I propose to have keys `block/HASH` and `undo/HASH`.
Also converting hashes to hex format is not necessary (because keys are bytestrings) and adds overhead, but on the other hand it makes debugging easier and overhead should be small, so I don't mind leaving it as is.
Also I would format hash to `Text` and then use `encodeUtf8`, because I think that `fromString` should mostly be used for conversion of string literals. And I would apply `undoModifier` after converting to `ByteString` (it should be slightly faster most likely). In this case strings are small and we know that `fromString` won't do anything subtle (because string will contain only ASCII characters), so I also don't mind leaving it as is.